regression/strings-smoke-tests/java_parselong/test_parselong.java

```diff
@@ -0,0 +1,39 @@
+public class test_parselong
+{
+    public static void main(String[] args)
+    {
+        if (args.length == 1) {
```

Comment:
Would the test run faster if they weren't all in the same file? Like presumably solving an equation with 5x more variables is more than 5x harder?